tools/fs/files.js

```diff
@@ -926,8 +940,13 @@ files.createTarGzStream = function (dirPath, options) {
       // setting it in an 'entry' handler is the same strategy that npm
       // does, so we do that here too.
       if (entry.type === "Directory") {
-        entry.mode = (entry.mode || entry.props.mode) | 0o500;
-        entry.props.mode = entry.mode;
+        entry.props.mode = addExecBitWhenReadBitPresent(entry.props.mode);
```

Comment:
First, I should point out that `entry.mode` is always `undefined` and never set, or used – maybe it was used in a previous version of `fstream`, but I can't find any (we use our [own fork](https://github.com/meteor/fstream/)), and this behavior matches the published `fstream` behavior.  The second parameter to the `filter` callback [is actually `props`](https://github.com/meteor/fstream/blob/cf4ea6c175355cec7bee38311e170d08c4078a5d/lib/reader.js#L180), so this would make sense in that scenario but this code doesn't do that, instead utilizing only the first argument which includes `props` as an attribute on `entry`.

For directories, I'd say that the `| 0o500` (e.g. `u+rx`) was never necessary and that something less would have accomplished the goal.  I'm happy to change that back, but my thinking was that it might be preferable to maintain some sense of permissions on Windows.  Essentially, if a directory was encountered that had the readable bit missing, we shouldn't simply add that read bit _and_ add the executable bit.  Presumably there would be some reason that `read` permission (something Windows does understand) was not set.  My train of thought was strongly influenced by that of the other PRs marked as closed and merged in [the `node-tar` issue](https://github.com/npm/node-tar/issues/7), for example, https://github.com/mapbox/node-pre-gyp/commit/7a28f4b0f562ba4712722fefe4eeffb7b20fbf7a and https://github.com/Microsoft/types-publisher/pull/165/commits/be208195326e2d6dab1f099a4989120f553251a5#diff-db71419dab88a2797059646c6a31a328R42.